### PR TITLE
Refactor Keithley drivers

### PR DESCRIPTION
**Pull request by jenshnielsen**
Convert all Keithley drivers to new add_parameter format.
Don't document inherited attributes
Document base classes
A fix docs fixes

TODO

- [x] News fragment

### CODE REVIEW

File: src/qcodes/instrument_drivers/Keithley/Keithley_3706A.py [hunk 1]
```diff
@@ -65,31 +65,49 @@ def __init__(
...
+If it is set to break before make,
+it is ensured that all channels open
+before any channels close.
+
+If it is set to make before break, it is
+ensured that all channels close before any
+channels open.
+
+If it is off, channels open and close
+simultaneously."""
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/Keithley_3706A.py:82]: Reformat

File: src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py [hunk 2]
```diff
@@ -374,32 +374,35 @@ def __init__(self, parent: Instrument, name: str, channel: str) -> None:
         vlimit_minmax = self.parent._vlimit_minmax
         ilimit_minmax = self.parent._ilimit_minmax
 
-        self.add_parameter(
+        self.volt: _MeasurementVoltageParameter = self.add_parameter(
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py:377]: These types should really be public

File: src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py [hunk 3]
```diff
@@ -475,8 +484,9 @@ def __init__(self, parent: Instrument, name: str, channel: str) -> None:
             "set `sourcerange_v` instead",
             vals=vals.Enum(*vranges[self.model]),
         )
+        """The range to perform voltage measurements in. This affects the range and the precision of the measurement. Note that if you both measure and source current this will have no effect, set `sourcerange_v` instead"""
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py:487]: reformat

File: src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py [hunk 4]
```diff
@@ -525,8 +538,9 @@ def __init__(self, parent: Instrument, name: str, channel: str) -> None:
             "`sourcerange_i` instead",
             vals=vals.Enum(*iranges[self.model]),
         )
+        """The range to perform current measurements in. This affects the range and the precision of the measurement. Note that if you both measure and source current this will have no effect, set `sourcerange_i` instead"""
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/_Keithley_2600.py:541]: reformat

File: src/qcodes/instrument_drivers/Keithley/Keithley_2400.py [hunk 5]
```diff
@@ -134,6 +147,7 @@ def __init__(
             "Note that it is an error to read current "
             "and voltage with output off",
         )
+        """Measure resistance from current and voltage Note that it is an error to read current and voltage with output off"""
```
Comment (astafan8) [inline src/qcodes/instrument_drivers/Keithley/Keithley_2400.py:150]:
```suggestion
        """Measure resistance from current and voltage. Note that it is an error to read current and voltage with output off"""
```

File: src/qcodes/instrument_drivers/Keithley/Keithley_2450.py [hunk 6]
```diff
@@ -306,14 +319,16 @@ def __init__(self, parent: "Keithley2450", name: str, proper_function: str) -> N
             "the internal reference measurements (autozero) of the"
             "instrument.",
         )
+        """This command enables or disables automatic updates tothe internal reference measurements (autozero) of theinstrument."""
```
Comment (astafan8) [inline src/qcodes/instrument_drivers/Keithley/Keithley_2450.py:322]:
```suggestion
        """This command enables or disables automatic updates to the internal reference measurements (autozero) of the instrument."""
```

**Review by astafan8 — APPROVED** [review]

File: src/qcodes/instrument_drivers/Keithley/Keithley_2450.py (hunk 6 above)
Comment (jenshnielsen) [reply src/qcodes/instrument_drivers/Keithley/Keithley_2450.py:322]: Do this manually to also resolve the docstr argument

File: src/qcodes/instrument_drivers/Keithley/Keithley_2400.py (hunk 5 above)
Comment (jenshnielsen) [reply src/qcodes/instrument_drivers/Keithley/Keithley_2400.py:150]: Do this manually to also resolve the docstr argument

**Pull request merged**